app/projects/steps/dynamic_form.py

```diff
@@ -0,0 +1,112 @@
+from abc import ABC, abstractmethod
+from projects.models import SignedAgreementForm
+from projects.models import AGREEMENT_FORM_TYPE_STATIC
+
+from projects.forms.payerdb import AccessRequestForm
+
+from projects.models import AgreementForm
+from projects.models import DataProject
+from projects.models import PayerDBForm
+
+from datetime import datetime
+
+class ProjectStep:
```

Comment:
I like what you did here with creating a class to represent the step and hold the necessary data in an OOD kind of way versus the dictionary -- more stable and reliable this way rather than an unenforced dictionary. I wonder if we could abstract this class a little further to replace all the step dictionaries that we still have in the `DataProjectView` class.

Reply:
This is more of a comment than a review...

Reply:
Actually that probably what you were going for right? Just seeing the SignAgreementFormsStepInitializer class now. Seems like we could move the agreement_form class variable to SignAgreementFormsStepInitializer and then this class would be good to go for using it with the other steps in the general project sign up process. Probably would move this class out of this particular file since it would no longer be "dynamic form" specific.

Reply:
Last comment -- i could take care of this stuff in a separate ticket if you agree since it's not a blocking feature. In the meantime, approving!

Reply:
I might try and tidy a few more things when I move on to saving the forms but then we can create tickets for the rest of the logic moving.